GitUI/UserControls/RevisionGrid/Graph/JunctionStyler.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections.Generic;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Linq;
+using GitCommands;
+
+namespace GitUI.UserControls.RevisionGrid.Graph
+{
+    internal sealed class JunctionStyler
+    {
+        private readonly IJunctionColorProvider _junctionColorProvider;
+        private readonly List<Color> _junctionColors = new List<Color>(capacity: 2);
+
+        public JunctionStyler(IJunctionColorProvider junctionColorProvider)
+        {
+            _junctionColorProvider = junctionColorProvider;
+        }
+
+        public Brush GetLaneBrush()
```

Comment:
@drewnoakes I couldn't write any tests for this method as my head goes into spin when I try to understand the logic... 
I think this method is to blame for #5243

Reply:
I'll take a look.